lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -449,16 +440,16 @@ def process_storage_command
       @device_commands << devs
     end
 
+    sig { returns(T::Array[Models::QemuHCKDevice]) }
     def load_devices
-      @device_infos = []
+      @devices_list.flatten!
+      @devices_list.compact!
```

Comment:
I'm not for moving these into `load_devices`. The method is to read the list of devices and load the devices, not to modify the list.

Reply:
As only `load_devices` uses `@devices_list` I preferred that it will fix all duplication/etc in `@devices_list`. Also, I want to simplify `initialize` where this is possible. Maybe call `@devices_list.flatten.compact.each` instead of updating the list?

Reply:
It is a good idea to push state changes that happen only at initialization time to `initialize`; it will clarify the order of state changes. The class is too complicated if that complicates `initialize` too much.

In this case, I believe `@devices_list` should be typed as `T::Array[String]`, and arrays or `nil` shouldn't be pushed into `@devices_list`, which ensures the array is always in a form suited for consuming. This also applies to other variables.

However, I don' request to make such a change now. Keeping the status quo by leaving `@devices_list.flatten!` and `@devices_list.compact!` in `initialize` is enough for this pull request.